ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/LedgerDB/StateMachine.hs

```diff
@@ -460,40 +465,35 @@ openLedgerDB flavArgs env cfg fs = do
 data Environment =
   Environment
     (LedgerDB' IO TestBlock)
-    (OnDemandActions' IO TestBlock)
```

Comment:
If you want to get rid of the `undefined`s, you could use `MVar`s here instead, which would initially be empty

Reply:
Right, but then I have to deal with them in every use site, no?